pandas/core/groupby/groupby.py

```diff
@@ -1098,7 +1098,8 @@ def reset_identity(values):
                 group_names = self.grouper.names
 
                 result = concat(values, axis=self.axis, keys=group_keys,
-                                levels=group_levels, names=group_names)
+                                levels=group_levels, names=group_names,
```

Comment:
same here, there are like 10 calls to concat. I think should be explicit about sort